tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -2212,3 +2214,84 @@ def benford_correlation(x):
     # np.corrcoef outputs the normalized covariance (correlation) between benford_distribution and data_distribution.
     # In this case returns a 2x2 matrix, the  [0, 1] and [1, 1] are the values between the two arrays
     return np.corrcoef(benford_distribution, data_distribution)[0, 1]
+
+
+@set_property("fctype", "combiner")
+def matrix_profile(x, param):
+    """
+    Calculates the 1-D Matrix Profile[1] and returns Tukey's Five Number Set plus the mean of that Matrix Profile.
+
+    .. rubric:: References
+
+    |  [1] Matrix Profile I: All Pairs Similarity Joins for Time Series: A Unifying View That Includes Motifs, Discords and Shapelets. Yeh et.al, 2016.
+
+    :param x: the time series to calculate the feature of
+    :type x: numpy.ndarray
+    :param param: contains dictionaries {"sample_pct": x, "threshold": y, "feature": z}
+    with sample_pct and threshold being parameters of the matrixprofile
+    package https://matrixprofile.docs.matrixprofile.org/api.html#matrixprofile-compute
+    and feature being one of "min", "max", "mean", "median", "25", "75"
+    and decides which feature of the matrix profile to extract
+    :type param: list
+    :return: the different feature values
+    :return type: pandas.Series
+    """
+
+    x = np.asarray(x)
+
+    def _calculate_mp(**kwargs):
+        """Calculate the matrix profile using the specified window, or the maximum subsequence if no window is specified"""
+        try:
+            if "windows" in kwargs:
+                m_p = mp.compute(x,**kwargs)['mp']
+
+            else:
+                m_p = mp.algorithms.maximum_subsequence(x, include_pmp=True,**kwargs)['pmp'][-1]
+
+            return m_p
+
+        except:
+            return [np.nan]
+
+    # The already calculated matrix profiles
+    matrix_profiles = {}
+
+    # The results
+    res = {}
+
+    for kwargs in param:
+        key = convert_to_output_format(kwargs)
+        feature = kwargs.pop('feature')
+
+        featureless_key = convert_to_output_format(kwargs)
+        if featureless_key not in matrix_profiles:
+            matrix_profiles[featureless_key] = _calculate_mp(**kwargs)
+
+        m_p = matrix_profiles[featureless_key]
+
+        #Set all features to nan if Matrix Profile is nan (cannot be computed)
+        if len(m_p) == 1:
```

Comment:
Can the len be 1 also for "normal" cases? If that can ever happen, I would propose to make this one simpler:
do not return `[np.nan]` on errors, but actually `None` and here only check for `if m_p is None`- I also think that is more pythonic, but that might be a matter of taste

Reply:
The good news is that the length cannot be 1 for normal cases, otherwise I'd definitely go with your approach.